src/legacy/ui/public/agg_response/point_series/_tooltip_formatter.js

```diff
@@ -47,8 +47,14 @@ export function PointSeriesTooltipFormatter($compile, $rootScope) {
         addDetail(data.xAxisLabel, data.xAxisFormatter(datum.x));
       }
       if (datum.y) {
+        const mode = event.config.get('valueAxes[0].scale.mode');
```

Comment:
we are using chart configuration inside response handler, which is not ideal.
preferably response handlers wouldn't know anything about the chart .. so they can work with different charts.
here you make an assumption that this tooltip handler will always be used from our bar/area/line chart implementation, which will have such setting.  Also this specific setting (config) will only be emited on hower (special case) and not with any other chart events.

Unfortunately we don't have a better (quick) way to fix it. Maybe just pass down this exact setting on the event, something like `isPercentageMode`, just to make it a bit more explicit and clear what we are doing.

Reply:
to do that check `ui/vislib/lib/dispatch.js` line 115

Reply:
Moved the logic to dispatch.js